src/reconcile/rec_write.c

```diff
@@ -569,17 +568,19 @@ __rec_write_check_complete(
 		return (EBUSY);
 
 	/*
-	 * Eviction can configure lookaside table reconciliation, consider if
-	 * it's worth giving up this reconciliation attempt and falling back to
-	 * using the lookaside table.  We continue with evict/restore if
-	 * switching to the lookaside doesn't make sense for any reason: we
-	 * won't retry an evict/restore reconciliation until/unless the
-	 * transactional system moves forward, so at worst it's a single wasted
-	 * effort.
+	 * Check if this reconciliation attempt is making progress.  If there's
+	 * any sign of progress, don't fall back to the lookaside table.
 	 *
-	 * First, check if the lookaside table is a possible alternative.
+	 * Check if the current reconciliation split, in which case we'll
+	 * likely get to write at least one of the blocks.  If we've created a
+	 * page image for a page that previously didn't have one, or we had a
+	 * page image and it is now empty, that's also progress.
 	 */
-	if (lookaside_retryp == NULL)
+	if (r->multi_next > 1)
+		return (0);
+	if (r->multi_next == 0 && r->page->dsk != NULL)
+		return (0);
+	if (r->multi_next == 1 && r->page->dsk == NULL)
```

Comment:
It would be useful to enhance this comment to say more about the paths being detected here -- I can't off-hand think of how we might have had a disk image and it would be gone without the page being discarded?